src/implementation.cpp

```diff
@@ -129,11 +129,12 @@ const implementation *detect_best_supported_implementation_on_first_use::set_bes
 
   if (force_implementation_name) {
     auto force_implementation = available_implementations[force_implementation_name];
-    if (!force_implementation) {
-      fprintf(stderr, "SIMDJSON_FORCE_IMPLEMENTATION environment variable set to '%s', which is not a supported implementation name!\n", force_implementation_name);
-      abort();
+    if (force_implementation) {
+      return active_implementation = force_implementation;
+    } else {
+      // Note: abort() and stderr usage within the library is forbidden.
+      return active_implementation = &unsupported_singleton;
```

Comment:
This is reasonable, but it would be nice if the error message you got when you used it told you that you tried to specify an invalid implementation name (and what it was).

Reply:
It is not really necessary to do that as part of this IMO. It makes the experience a little worse, but it's an internal-developer-only situation generally.

Reply:
Right. I don't know how to do that, but I figured that getting an unsupported error would be a good hint. Note that relying on system variables is really hell from a usability point of view. You know how you set the variable and then forget you set it and then things break? I would argue that relying on system variable is bad practice. But we can fight about it in another PR.

Reply:
(Think about debugging a user who has this variable set somehow... how long would it take before you asked the user about his system variables?)

Reply:
Just wanted to respond to the system variables thing ... I tend to see this as a dev-internal thing and not something we should expose. Maybe we should add a dev only variable for it, or something, but it would be nice to be able to tell users to set this and tell us what happens for different implementations. A compromise would be to have json2json -v print the fact that the environment variable is set alongside the implementation chosen.